lib/src/server.dart

```diff
@@ -170,22 +173,21 @@ class Server {
         'result': result,
         'id': request['id']
       };
-    }).catchError((error, stackTrace) {
-      if (error is! RpcException) {
-        error = new RpcException(
-            error_code.SERVER_ERROR, getErrorMessage(error), data: {
-          'full': error.toString(),
-          'stack': new Chain.forTrace(stackTrace).toString()
-        });
-      }
-
+    } catch (error, stackTrace) {
       if (error.code != error_code.INVALID_REQUEST &&
```

Comment:
`error.code` may not be defined here, since we don't know anything about the type of `error` yet. I meant just pulling the `request.containsKey()` check here, and leaving the `error.code` check where it was.